components/constellation/constellation.rs

```diff
@@ -1199,13 +1242,12 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         // Notify the browser chrome that the pipeline has failed
         self.trigger_mozbrowsererror(top_level_frame_id, reason, backtrace);
 
-        let frame_id = FrameId::from(top_level_frame_id);
-        let pipeline_id = self.frames.get(&frame_id).map(|frame| frame.current.pipeline_id);
+        let pipeline_id = self.frames.get(&top_level_frame_id).map(|frame| frame.current.pipeline_id);
         let pipeline_url = pipeline_id.and_then(|id| self.pipelines.get(&id).map(|pipeline| pipeline.url.clone()));
         let parent_info = pipeline_id.and_then(|id| self.pipelines.get(&id).and_then(|pipeline| pipeline.parent_info));
         let window_size = pipeline_id.and_then(|id| self.pipelines.get(&id).and_then(|pipeline| pipeline.size));
 
-        self.close_frame_children(frame_id, ExitPipelineMode::Force);
+        self.close_frame_children(top_level_frame_id, ExitPipelineMode::Force);
```

Comment:
Once we've got document discarding, that should all happen automatically, because the old page will be discarded, which will remove the script channel map entry.